tests-integration/tests/http.rs

```diff
@@ -1542,7 +1542,52 @@ transform:
     )
     .await;
 
-    // 4. remove pipeline
+    // 4. cross-ref pipeline
+    // create database test_db
+    let res = client
+        .post("/v1/sql?sql=create database test_db")
+        .header("Content-Type", "application/x-www-form-urlencoded")
+        .send()
+        .await;
+    assert_eq!(res.status(), StatusCode::OK);
+
+    // check test_db created
+    validate_data(
+        "pipeline_db_schema",
+        &client,
+        "show databases",
+        "[[\"greptime_private\"],[\"information_schema\"],[\"public\"],[\"test_db\"]]",
+    )
+    .await;
+
+    // should be failed if not specify db
+    let res = client
+        .post("/v1/ingest?db=test_db&table=logs1&pipeline_name=test")
+        .header("Content-Type", "application/json")
+        .body(data_body)
+        .send()
+        .await;
+    assert_eq!(res.status(), StatusCode::BAD_REQUEST);
+
+    // cross ref using public's pipeline
+    let res = client
```

Comment:
Should pipeline permissions for cross-database access be checked, or does the database access plugin already handle this? If not, should we intercept the request?